include/circt/Dialect/FIRRTL/FIRRTLCanonicalization.td

```diff
@@ -136,37 +125,37 @@ def GetWidthAsIntAttr : NativeCodeCall<
                     "$0.getType().cast<FIRRTLType>().getBitWidthOrSentinel())">;
 
 ////////////////////////////////////////////////////////////////////////////////
-// DontTouch application
+// Drop name if no symbol (DontTouch application)
 ////////////////////////////////////////////////////////////////////////////////
 
 // node(x,name),!dnt -> node(x, "")
 def DropNameNode : Pat<
   (NodeOp $x, $name, $annotations, $inner_sym),
   (NodeOp $x, (GetEmptyString), $annotations, $inner_sym),
-  [(uselessNameAttr $name), (NoDontTouch $annotations)]>;
+  [(uselessNameAttr $name), (NullAttr $inner_sym)]>;
 
 // wire(name),!dnt -> wire("")
 def DropNameWire : Pat<
   (WireOp $name, $annotations, $inner_sym),
   (WireOp (GetEmptyString), $annotations, $inner_sym),
-  [(uselessNameAttr $name), (NoDontTouch $annotations)]>;
+  [(uselessNameAttr $name), (NullAttr $inner_sym)]>;
 
 // reg(clk, name),!dnt -> reg(clk, "")
 def DropNameReg : Pat<
   (RegOp $clk, $name, $annotations, $inner_sym),
   (RegOp $clk, (GetEmptyString), $annotations, $inner_sym),
-  [(uselessNameAttr $name), (NoDontTouch $annotations)]>;
+  [(uselessNameAttr $name), (NullAttr $inner_sym)]>;
 
 // regreset(clk, sig, val, name),!dnt -> regreset(clk, sig, val, "")
 def DropNameRegReset : Pat<
   (RegResetOp $clk, $sig, $val, $name, $annotations, $inner_sym),
   (RegResetOp $clk, $sig, $val, (GetEmptyString), $annotations, $inner_sym),
-  [(uselessNameAttr $name), (NoDontTouch $annotations)]>;
+  [(uselessNameAttr $name), (NullAttr $inner_sym)]>;
 
 // Mem(...,name),!dnt -> Mem(...x, "")
 def DropNameMem : Pat<
   (MemOp $rd, $rw, $depth, $ruw, $portNames, $name, $annotations, $portAnno, $inner_sym),
   (MemOp $rd, $rw, $depth, $ruw, $portNames, (GetEmptyString), $annotations, $portAnno, $inner_sym),
-  [(uselessNameAttr $name), (NoDontTouch $annotations), (NoPortsDontTouch $portAnno)]>;
```

Comment:
inner_sym only applies to the memory as a whole, right?  The ports still need to be checked?